auth/src/main/java/com/firebase/ui/auth/util/CredentialsUtils.java

```diff
@@ -0,0 +1,69 @@
+package com.firebase.ui.auth.util;
+
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+import android.text.TextUtils;
+
+import com.firebase.ui.auth.IdpResponse;
+import com.google.android.gms.auth.api.credentials.Credential;
+import com.google.firebase.auth.FirebaseUser;
+
+/**
+ * Utility class for working with {@link Credential} objects.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class CredentialsUtils {
+    private CredentialsUtils() {
+        throw new AssertionError("No instance for you!");
+    }
+
+    /**
+     * Build a credential for the specified {@link FirebaseUser} with optional password and {@link
+     * IdpResponse}.
+     * <p>
+     * If the credential cannot be built (for example, empty email) then will return {@code null}.
+     */
+    @Nullable
+    public static Credential buildCredential(@NonNull FirebaseUser user,
+                                             @Nullable String password,
+                                             @Nullable String accountType) {
+        String email = user.getEmail();
+        String phone = user.getPhoneNumber();
+        Uri profilePictureUri =
+                user.getPhotoUrl() == null ? null : Uri.parse(user.getPhotoUrl().toString());
+
+        if (TextUtils.isEmpty(email) && TextUtils.isEmpty(phone)
```

Comment:
Nit: this would be clearer if separated into two if-return-null statements.

if (no phone and no email) {
  return null;
}

if (no password and no account type) {
  return null
}

Reply:
No probs.